nemo/core/connectors/save_restore_connector.py

```diff
@@ -0,0 +1,325 @@
+    ):
+        """
+		Restores model instance (weights and configuration) into .nemo file
+		Args:
+		restore_path: path to .nemo file from which model should be instantiated
+		override_config_path: path to a yaml config that will override the internal
+			config file or an OmegaConf / DictConfig object representing the model config.
+		map_location: Optional torch.device() to map the instantiated model to a device.
+			By default (None), it will select a GPU if available, falling back to CPU otherwise.
+		strict: Passed to load_state_dict. By default True
+		return_config: If set to true, will return just the underlying config of the restored
+			model as an OmegaConf DictConfig object without instantiating the model.
+
+		Example:
+			```
+			model = nemo.collections.asr.models.EncDecCTCModel.restore_from('asr.nemo')
+			assert isinstance(model, nemo.collections.asr.models.EncDecCTCModel)
+			```
+
+		Returns:
+		An instance of type cls or its underlying config (if return_config is set).
+		"""
+        app_state = AppState()
+
+        # Get path where the command is executed - the artifacts will be "retrieved" there
+        # (original .nemo behavior)
+        cwd = os.getcwd()
+
+        if map_location is None:
+            if torch.cuda.is_available():
+                map_location = torch.device('cuda')
+            else:
+                map_location = torch.device('cpu')
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            try:
+                self._unpack_nemo_file(path2file=restore_path, out_folder=tmpdir)
+                os.chdir(tmpdir)
+                if override_config_path is None:
+                    config_yaml = path.join(tmpdir, app_state.model_config_yaml)
+                else:
+                    # can be str path or OmegaConf / DictConfig object
+                    config_yaml = override_config_path
+                if not isinstance(config_yaml, (OmegaConf, DictConfig)):
+                    conf = OmegaConf.load(config_yaml)
+                else:
+                    conf = config_yaml
+                    if override_config_path is not None:
+                        # Resolve the override config
+                        conf = OmegaConf.to_container(conf, resolve=True)
+                        conf = OmegaConf.create(conf)
+                # If override is top level config, extract just `model` from it
+                if 'model' in conf:
+                    conf = conf.model
+
+                if return_config:
+                    instance = conf
+                else:
+                    app_state = AppState()
+                    if app_state.model_parallel_rank is not None:
+                        model_weights = path.join(
+                            tmpdir, f'mp_rank_{app_state.model_parallel_rank:02}', app_state.model_weights_ckpt
+                        )
+                    else:
+                        model_weights = path.join(tmpdir, app_state.model_weights_ckpt)
```

Comment:
This seems to be related with model parallel handling. Is there any chance to "abstract" that and move to a separate method, so if TLT will overload `restore_from` it will be able to inherit and leverage that logic?

Reply:
Added a method. Please let me know if was what you had in mind.